controllers/secret_controller.go

```diff
@@ -206,6 +229,49 @@ func (r *SecretReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 	}
 }
 
+// usernameFromNode identifies the username associated with a given node
+func (r *SecretReconciler) usernameFromNode(ctx context.Context, node *core.Node, nodes *core.NodeList) (string, error) {
```

Comment:
I find this function construction to be a bit strange. You are passing a `node` and a list of nodes to get the username from the node. This is hinting that `ParseInstances` needs to be refactored better. Moreover why do you need to cross-reference the list of nodes with the `ConfigMap` here?

Reply:
The node list being passed in 1. to then pass along to `parseInstances` and 2. just to reduce the amount of API calls (since we already have the node list at the site of this function call). I agree this is a bit strange for now. Improvement can be made based on the `windows-instances parser pkg that will be introduced in #499, but I am not clear how I could better this in this PR currently.

Reply:
>Moreover why do you need to cross-reference the list of nodes with the ConfigMap here?

You still have not answered this. Moreover having `usernameFromNode` as part of the `SecretReconciler` is also strange. I suggest pairing with @sebsoto to find a better solution for this.

Reply:
@sebsoto and I refactored the logic here. We moved the username retrieval to a helper `getNodeUsername` no longer associated with the secret controller.